lib/ProductOpener/Store.pm

```diff
@@ -86,6 +87,10 @@ sub get_string_id_for_lang {
 	if (not defined $string) {
 		return "";
 	}
+	
+	# Normalize Unicode characters
+	# Form NFC
+	$string = NFC($string); 
```

Comment:
Am I correct in the assumption that the `NFC` sub uses native code instead of pure perl, so that the performance should be quite good? I'm just wary of using a 3rd party method here, since I seem to remember the performance impact of changes in this method was quite important for overall ProductOpener performance. 🙂

Reply:
> Am I correct in the assumption that the NFC sub uses native code instead of pure perl

Correct. [It's implemented in XS](https://metacpan.org/source/KHW/Unicode-Normalize-1.26/Normalize.xs#L576)

Reply:
I did some testing, running get_string_id_for_lang 1M times with a small string:

Original function:
root@ns3135968:/srv/off/scripts# time ./test_get_string_id.pl 

real	0m12.530s
user	0m12.524s
sys	0m0.004s

---

Regexp approach:
root@ns3135968:/srv/off/scripts# time ./test_get_string_id_regexp.pl 

real	0m15.718s
user	0m15.704s
sys	0m0.012s

---
With Unicode::Normalize:
root@ns3135968:/srv/off/scripts# time ./test_get_string_id_nfc.pl 

real	0m15.754s
user	0m15.744s
sys	0m0.008s

I think it's worth using Unicode::Normalize, as we actually need to normalize to match entries to the taxonomies (in all languages with potentially composed characters like accents, not only when we want to remove accents).

Reply:
> Correct. [It's implemented in XS](https://metacpan.org/source/KHW/Unicode-Normalize-1.26/Normalize.xs#L576)

Perfect, thanks. 🙂 

> I think it's worth using Unicode::Normalize, as we actually need to normalize to match entries to the taxonomies (in all languages with potentially composed characters like accents, not only when we want to remove accents).

Agreed, and the benchmarks don't look like the impact is too bad. Thanks for adding the benchmark numbers. 🙂